clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/AutoElasticsearchDocumentMapper.scala

```diff
@@ -48,8 +54,11 @@ class AutoElasticsearchDocumentMapper[ModelKey <: Product: ClassTag,
 object AutoElasticsearchDocumentMapper {
   def apply[ModelKey <: Product: ClassTag,
             ModelMetadata: ClassTag,
-            Document: ClassTag]
+            Document <: ClioDocument: ClassTag]
     : ElasticsearchDocumentMapper[ModelKey, ModelMetadata, Document] = {
-    new AutoElasticsearchDocumentMapper
+    // Don't listen to IntelliJ's warning on the line below; removing
+    // the _ results in a compiler deprecation warning, which our settings
+    // escalate into an error.
+    new AutoElasticsearchDocumentMapper(ClioUUIDGenerator.getUUID _)
```

Comment:
You could re-define ClioUUIDGenerator.getUUID as a function instead of a method.
Then you don't need to lift it when passing it as a value.

object ClioUUIDGenerator {
  private val generator: TimeBasedGenerator =
    Generators.timeBasedGenerator(EthernetAddress.fromInterface())
  def getUUID = () => generator.generate
}

But I guess I don't care either way. Thank you, Scala.